packages/backend/src/modules/tvl2/PriceIndexer.ts

```diff
@@ -0,0 +1,142 @@
+import { assert, Logger } from '@l2beat/backend-tools'
+import { CoingeckoQueryService } from '@l2beat/shared'
+import { PriceConfigEntry, UnixTime } from '@l2beat/shared-pure'
+import { ChildIndexer } from '@l2beat/uif'
+import { Knex } from 'knex'
+
+import { IndexerStateRepository } from '../../peripherals/database/repositories/IndexerStateRepository'
+import { HourlyIndexer } from '../liveness/HourlyIndexer'
+import { PricesRecord, PricesRepository } from './repositories/PricesRepository'
+import { SyncOptimizer } from './SyncOptimizer'
+
+export class PriceIndexer extends ChildIndexer {
+  indexerId: string
+
+  constructor(
+    logger: Logger,
+    parentIndexer: HourlyIndexer,
+    private readonly coingeckoQueryService: CoingeckoQueryService,
+    private readonly stateRepository: IndexerStateRepository,
+    private readonly pricesRepository: PricesRepository,
+    private readonly token: PriceConfigEntry,
```

Comment:
I would actually see this done slightly differently. The PriceIndexer would receive a single coingeckoId and multiple PriceConfigEntries with this coingeckoId set. Then it will start downloading data from the min of those entries timestamps and then save to the entries that need it. That way we're not redownloading data for the same coingeckoId.